packages/backend/src/database/migrations/20220222155435_alter_chart_configuration.ts

```diff
@@ -0,0 +1,96 @@
+import { Knex } from 'knex';
+
+export async function up(knex: Knex): Promise<void> {
+    // Add new chart type
+    await knex('chart_types').insert({ chart_type: 'cartesian' });
+
+    // Add new column for chart config and pivot config
+    await knex.schema.alterTable('saved_queries_versions', (tableBuilder) => {
+        tableBuilder.jsonb('chart_config').nullable();
+        tableBuilder.specificType('pivot_dimensions', 'TEXT[]').nullable();
+    });
+
+    // Migrate configs for big_number and table
+    await knex.raw(`
+    UPDATE saved_queries_versions
+    SET chart_config = '{}'::jsonb
+    WHERE chart_type in ('big_number', 'table')`);
+
+    // Migrate pivot_config, chart_type and chart_config for bar, line, scatter, column
+    await knex.raw(`WITH updates AS (
+        SELECT 
+            sqv.saved_queries_version_id, 
+            json_build_object(
+                'series', (
+                    CASE
+                        WHEN sqv.x_dimension IS NULL THEN '[]'::jsonb
+                        ELSE COALESCE((
+                            select jsonb_agg(jsonb_build_object(
+                                'type', 
+                                    CASE
+                                        WHEN sqv.chart_type in ('bar', 'column') THEN 'bar'
+                                        ELSE sqv.chart_type
+                                    END,
+                                'yField', ym.field_name,
+                                'xField', sqv.x_dimension,
+                                'flipAxes', sqv.chart_type = 'bar'
+                            )) 
+                            from saved_queries_version_y_metrics ym 
+                            where ym.saved_queries_version_id = sqv.saved_queries_version_id
+                        ), '[]'::jsonb)
+                    END
+                )
+            ) as chart_config
+        FROM saved_queries_versions sqv
+        WHERE chart_type not in ('big_number', 'table')
+    ) 
+    
+    UPDATE saved_queries_versions sqv
+    SET chart_type = 'cartesian',
+        pivot_dimensions = (
+            CASE 
+                WHEN sqv.group_dimension is NULL THEN NULL
+                ELSE ARRAY[sqv.group_dimension]::TEXT[] 
+            END
+        ),
+        chart_config = updates.chart_config
+    FROM updates
+    WHERE sqv.saved_queries_version_id = updates.saved_queries_version_id
+        AND chart_type not in ('big_number', 'table')
+    `);
+
+    // Delete unused y_metrics table
+    await knex.schema.dropTableIfExists('saved_queries_version_y_metrics');
```

Comment:
We need to recreate this table on rollback

Reply:
Working.

rollback on latest migration + switch to `main` and then managed to create charts. also dashboards with references to deleted charts still work as expected: 
<img width="1537" alt="Screenshot 2022-02-24 at 10 38 08" src="https://user-images.githubusercontent.com/11660098/155508474-d7f76cfc-9f88-4556-86c2-bcbe8b7980dc.png">